src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -50,5 +49,13 @@ public SNIMarsConnection GetConnection(SNIHandle lowerHandle)
         {
             return _connections[lowerHandle];
         }
+
+        /// <summary>
+        /// Clears all Mars connections managed by this class.
+        /// </summary>
+        public void ClearMarsConnections()
+        {
+            _connections.Clear();
```

Comment:
Clearing was sufficient to fix the memory accumulation issue, since we were just accumulating objects the code was never going to touch again. I'll check to see if disposing the underlying MarsHandles it references is necessary here. I'm not sure if they would have all been disposed of already before this point.